plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -266,8 +275,41 @@ private Client client() {
     @FunctionalInterface
     interface ProcessorFactoryProvider {
 
-        Processor.Factory get(Environment environment);
+        Processor.Factory get(Environment environment, TemplateService templateService);
+
+    }
+
+    static class InternalTemplateService implements TemplateService {
+
+        public static final ScriptContext.Plugin INSTANCE = new ScriptContext.Plugin("elasticsearch-ingest", "ingest");
+
+        private final ScriptService scriptService;
+
+        InternalTemplateService(ScriptService scriptService) {
+            this.scriptService = scriptService;
+        }
 
+        @Override
+        public Template innerCompile(String template) {
+            Script script = new Script(template, ScriptService.ScriptType.INLINE, "mustache", Collections.emptyMap());
+            CompiledScript compiledScript = scriptService.compile(script, INSTANCE, null /* only needed for indexed scripts */);
```

Comment:
no, it means the null can be specified because ingest isn't using indexed scripts. I'll update the comment :)